cloud-azure/build.gradle

```diff
@@ -37,6 +37,7 @@ dependencies {
   implementation (group: 'com.microsoft.azure',         name: 'azure-mgmt-sql',             version: azureSdkVersion) { exclude group: 'org.slf4j' }
   implementation (group: 'com.microsoft.azure.privatedns.v2018_09_01',         name: 'azure-mgmt-privatedns',             version: '1.0.0-beta') { exclude group: 'org.slf4j' }
   implementation group: 'com.microsoft.azure',         name: 'azure-storage',              version: azureStorageSdkVersion
+  implementation (group: 'com.azure.resourcemanager',         name: 'azure-resourcemanager-keyvault',        version: '2.13.0')
```

Comment:
The method `isRolesBasedAccessControl` is available in version 2.x
https://github.com/hortonworks/cloudbreak/pull/12629/files#diff-05e39f47074b19cbc8da8a8b13d930dbbd8fc0bc3bc408d146e9625482a58ad8R1032

What else I can explore ?

Reply:
com.azure.* (version 2.x) is required to use method `isRolesBasedAccessEnabled` - to check if keyVault has roles based access. Without this check assigning the access would fail. 

As there are compatibility issues, A solution I can think of is - 
Take an extra parameter in the `create-azure-environment` CLI, `--is-role-based-access-enabled-for-keyVault`. Customer would provide this parameter along with encryption key only if keyVault is roles based. This param would replace `isRoleBasedAccessEnabled` method call to Azure and eliminate the need of 2.x version.

Reply:
I have added a new param `isRoleBasedAccessControlEnabled` to `AzureResourceEncryptionParameters` to resolve the sdk version issue.